pkg/apis/pipeline/v1beta1/result_defaults_test.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2022 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/test/diff"
+)
+
+func TestTaskResult_SetDefaults(t *testing.T) {
+	tests := []struct {
+		name   string
+		before *v1beta1.TaskResult
+		after  *v1beta1.TaskResult
+	}{{
+		name: "inferred string type",
+		before: &v1beta1.TaskResult{
+			Name: "resultname",
+		},
+		after: &v1beta1.TaskResult{
+			Name: "resultname",
+			Type: v1beta1.ResultsTypeString,
```

Comment:
Can you also add some test cases where result type is provided? i.e. if the user specified `string` type, the expected is string type; if `array` type is specified, expected is `array` etc.

Reply:
sure!